features/admin.applications.v1/components/settings/general-application-settings.tsx

```diff
@@ -226,6 +229,60 @@ export const GeneralApplicationSettings: FunctionComponent<GeneralApplicationSet
                     message: t("applications:notifications.updateApplication.genericError" +
                         ".message")
                 }));
+            });
+    };
+
+    /**
+     * Handles the toggle change to show confirmation modal.
+     *
+     * @param event - Form event.
+     * @param data - Checkbox data.
+     */
+    const handleAppEnableDisableToggleChange = (event: FormEvent<HTMLInputElement>, data: CheckboxProps): void => {
+        setEnableStatus(data.checked);
+        setShowDisableConfirmationModal(true);
+    };
+
+
+    /**
+     * Disables an application.
+     */
+    const handleApplicationDisable = () => {
+        setIsDisableInProgress(true);
+        const state : string = enableStatus ? "enabled" : "disabled";
+        const state2 : string = enableStatus ? "enable" : "disable";
+
+        disableApplication(appId, enableStatus)
+            .then(() => {
+                setIsDisableInProgress(false);
```

Comment:
What happens if `disableApplication` throws an error?

`isDiasableProgress` will remain true right? Is that expected?

Reply:
Fixed it. Please check